src/extension/utils.js

```diff
@@ -143,6 +143,81 @@ export async function getState(cb) {
   }
 }
 
+/**
+ * Encodes the url to be used in the `/shares/` msgraph API call
+ * @param {string} sharingUrl
+ * @returns {string}
+ */
+export function encodeSharingUrl(sharingUrl) {
+  const base64 = btoa(sharingUrl)
+    .replace(/=/, '')
+    .replace(/\//, '_')
+    .replace(/\+/, '-');
+  return `u!${base64}`;
+}
+
+/**
+ * Fetches the sharepoint edit info directly from sharepoint
+ * @todo also use fstab information to figure out the resource path etc.
+ *
+ * @param {string} tabUrl
+ * @returns {Promise<object>}
+ */
+export async function fetchSharePointEditInfo(tabUrl) {
+  const shareLink = encodeSharingUrl(tabUrl);
+  const spUrl = new URL(tabUrl);
+  spUrl.pathname = `/_api/v2.0/shares/${shareLink}/driveItem`;
+  let resp = await fetch(spUrl);
+  if (!resp.ok) {
+    // eslint-disable-next-line no-console
+    console.warn('unable to resolve edit url: ', resp.status);
+    return null;
+  }
+  const data = await resp.json();
+
+  // get root item
+  spUrl.pathname = `/_api/v2.0/drives/${data.parentReference.driveId}`;
+  resp = await fetch(spUrl);
+  if (!resp.ok) {
+    // eslint-disable-next-line no-console
+    console.warn('unable to load root url: ', resp.status);
+    return null;
+  }
+  const rootData = await resp.json();
+
+  const info = {
+    status: 200,
+    name: data.name,
+    sourceLocation: `onedrive:/drives/${data.parentReference.driveId}/items/${data.id}`,
+    lastModified: data.lastModifiedDateTime,
+  };
+  if (data.folder) {
+    info.url = data.webUrl;
+    info.contentType = 'application/folder';
+    info.childCount = data.folder.childCount;
+  } else {
+    const folder = data.parentReference.path.split(':').pop();
+    info.url = `${rootData.webUrl}${folder}/${data.name}`;
+    info.contentType = data.file.mimeType;
+  }
+  return info;
+}
+
+export async function fetchGoogleEditInfo(tabUrl) {
+  // todo: fetch from gdrive ?
+  return {
+    url: tabUrl,
+  };
+}
+
+export function isSharePointHost(host) {
+  return /^[a-z-]+\.sharepoint\.com$/.test(host);
+}
+
+export function isGoogleDriveHost(host) {
+  return /(docs|drive)\.google\.com$/.test(host);
```

Comment:
You're right, let's add this for safety
```suggestion
  return /^(docs|drive)\.google\.com$/.test(host);
```